lib/rules/no-capital-arguments.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+const Rule = require('./base');
+
+const ERROR_MESSAGE = 'Capital argument names is not supported';
+const AllowedPrefix = /[a-z]/;
+
+module.exports = class NoCapitalArguments extends Rule {
+  visitor() {
+    return {
+      AttrNode(node) {
+        if (node.name.startsWith('@')) {
+          let firstChar = node.name.charAt(1);
+          if (!AllowedPrefix.test(firstChar)) {
```

Comment:
This isn't quite good enough. The current list of disallowed names can be found here:

https://github.com/emberjs/ember.js/blob/3a7c034cdc81c3ed05508f64c630e2041995d2f8/packages/ember-template-compiler/lib/plugins/assert-reserved-named-arguments.ts#L39-L43

```js
const RESERVED = ['@arguments', '@args', '@block', '@else'];

function isReserved(name: string): boolean {
  return RESERVED.indexOf(name) !== -1 || Boolean(name.match(/^@[^a-z]/));
}
```